test/Microsoft.AspNetCore.Authentication.Test/CookieTests.cs

```diff
@@ -1114,6 +1116,51 @@ private Task SignOutAsWrong(HttpContext context)
             Assert.Equal("Alice", FindClaimValue(transaction5, ClaimTypes.Name));
         }
 
+        [Fact]
+        public async Task CookieIsRenewedWithSlidingExpirationWithoutTransformations()
+        {
+            var server = CreateServer(o =>
+            {
+                o.ExpireTimeSpan = TimeSpan.FromMinutes(10);
+                o.SlidingExpiration = true;
+                o.Events.OnValidatePrincipal = c =>
+                {
+                    // https://github.com/aspnet/Security/issues/1607
+                    // On sliding refresh the transformed principal should not be serialized into the cookie, only the original principal.
+                    Assert.Single(c.Principal.Identities);
+                    Assert.True(c.Principal.Identities.First().HasClaim("marker", "true"));
```

Comment:
Are they still able to manipulate the principal that gets serialized via some other mechanism?

Reply:
Through OnValidatePrincipal like we do atm. And if that's not enough it might need a new event like OnTicketResponse which gets called by the Response.OnStarting handler that writes out the principal.

Reply:
Right, only OnValidatePrincipal is the only touch point where they can modify a deserialized user in such a way that it will be persisted, and only if they set ShouldRenew.